src/webportal/src/app/components/auto-complete.jsx

```diff
@@ -32,57 +47,50 @@ function getSpacingValue(unitString) {
   return numericalPart;
 }
 
-export const AutoComplete = ({items, value, onChange, showAllSuggestions}) => {
+export const AutoComplete = ({
+  items,
+  value,
+  onChange,
+  showAllSuggestions,
+}) => {
   const [focused, setFocused] = useState(false);
   const [suggestions, setSuggestions] = useState([]);
   const [suggested, setSuggested] = useState(0);
   const root = useRef();
   const input = useRef();
   // get callout's gap space
-  const {spacing} = getTheme();
-  const gap = useMemo(
-    () => getSpacingValue(spacing.s1),
-    [spacing],
-  );
+  const { spacing } = getTheme();
+  const gap = useMemo(() => getSpacingValue(spacing.s1), [spacing]);
   // suggestions
-  useEffect(
-    () => {
-      const suggestions = showAllSuggestions ? (
-        items
-      ) : (
-        items.sort().filter((x) => x.startsWith(value) && x !== value)
-      );
-      setSuggestions(suggestions);
-      if (suggested >= suggestions.length) {
-        setSuggested(Math.max(0, suggestions.length - 1));
-      }
-    },
-    [value, items, showAllSuggestions],
-  );
+  useEffect(() => {
+    const suggestions = showAllSuggestions
+      ? items
+      : items.sort().filter(x => x.startsWith(value) && x !== value);
+    setSuggestions(suggestions);
+    if (suggested >= suggestions.length) {
+      setSuggested(Math.max(0, suggestions.length - 1));
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
you should ignore this rule globally if you do not want to fix

Reply:
Currently I nearly could not fix them all unless discuss with authors about the purpose and process of each hook used. It will be a time-consuming work, but we hope they will be fixed in the future since the incorrect usage of hooks will cause [potential issues](https://reactjs.org/docs/hooks-rules.html).

And future hooks will receive this warning to remind developers to set dependencies with care.

Reply:
so I think exclude this rule in eslintrc will be better than add eslint-disable comments, you can change the rule back after fix them in the future

we should limit the use of eslint-disable, otherwise the lint will become useless

Reply:
disable this rule in temp and will reopen in future pr